orderer/common/channelparticipation/restapi.go

```diff
@@ -9,21 +9,35 @@ package channelparticipation
 import (
 	"encoding/json"
 	"net/http"
+	"path"
+	"strings"
 
+	"github.com/gorilla/mux"
+	"github.com/hyperledger/fabric/common/configtx"
 	"github.com/hyperledger/fabric/common/flogging"
 	"github.com/hyperledger/fabric/orderer/common/localconfig"
+	"github.com/hyperledger/fabric/orderer/common/types"
 	"github.com/pkg/errors"
 )
 
-const URLBaseV1 = "/participation/v1/"
-
-type ErrorResponse struct {
-	Error string `json:"error"`
-}
+const (
+	URLBaseV1           = "/participation/v1/"
+	URLBaseV1Channels   = URLBaseV1 + "channels"
+	channelIDKey        = "channelID"
+	urlWithChannelIDKey = URLBaseV1Channels + "/{" + channelIDKey + "}"
+)
 
 //go:generate counterfeiter -o mocks/channel_management.go -fake-name ChannelManagement . ChannelManagement
 
 type ChannelManagement interface {
+	// ListAll returns the names of all channels (including the system channel), and the name of the system channel
+	// (empty if does not exist). The URL field is empty, and is to be completed by the caller.
+	ListAllChannels() *types.ChannelList
+
+	// ListChannel provides extended status information about a channel.
+	// The URL field is empty, and is to be completed by the caller.
+	ListChannel(channelID string) (*types.ChannelInfo, error)
```

Comment:
why not concrete type?

Reply:
This is borderline, I agree. A non-pointer would work, since it is a "smallish" struct. However, i do return nil when the channel is not found.

Reply:
if you return an error in this case then it doesn't matter if you return nil if it's not found, or a zero value

Reply:
non- pointer type it is, then.